spec/unit/honeybadger/config/ruby_spec.rb

```diff
@@ -81,6 +81,33 @@
     end
   end
 
+  describe "#before_notify" do
+    it "adds a block as a before hook" do
+      block = ->(_notice) {}
+
+      subject.before_notify(&block)
+
+      expect(subject.to_hash).to eq(before_notify: [block])
+    end
+
+    it "adds a callable as a before hook" do
+      callable = ->(_notice) {}
+
+      subject.before_notify(callable)
+
+      expect(subject.to_hash).to eq(before_notify: [callable])
+    end
+
+    it "gives access to the before hooks when passed nothing" do
+      expect(subject.before_notify).to eq([])
+
+      callable = ->(_notice) {}
+      subject.before_notify(callable)
+
+      expect(subject.before_notify).to eq([callable])
+    end
```

Comment:
It'd be great if we can add a couple of tests to make sure errors are reported even if hooks throw an exception.